pkg/kafkaclient/client.go

```diff
@@ -27,7 +27,8 @@ import (
 )
 
 var log = logf.Log.WithName("kafka_util")
-var apiVersion = sarama.V2_6_0_0
+var apiVersion = sarama.MaxVersion
```

Comment:
According to https://github.com/Shopify/sarama/blob/v1.36.0/config.go#L455

```
// The version of Kafka that Sarama will assume it is running against.
	// Defaults to the oldest supported stable version. Since Kafka provides
	// backwards-compatibility, setting it to a version older than you have
	// will not break anything, although it may prevent you from using the
	// latest features. Setting it to a version greater than you are actually
	// running may lead to random breakage.
	Version KafkaVersion
```

This means that using `sarama.MaxVersion`  api version which points to `V3_2_0_0` may cause  to random issues in koperator when interacting with older Kafka versions (e.g. 2.6).

Reply:
Yeah in this case I support leaving it on the older version for backward compatibility reasons.
If this is a blocker for you Adrian, let us know and we will think about what else we could do.

My apologies for missing this.

Reply:
Kafka clients since 1.x are also backward compatible with older versions of brokers - in the sense that they negotiate the latest API version they can talk during the initial handshake. I believe that comment is a very old one (6 yrs back https://github.com/Shopify/sarama/commit/5e8b6ff817d9cde8fc3f9a8ba1201280de612bd8) when that was not the case.

Either way - I don't have a strong opinion and I can revert back this.
The intent was to ensure we always use the latest API version sarama client can talk and not having to worry about in the feature about bumping this in needed.